src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,274 @@
+      useNativeDriver = true,
+      onPress,
+      ...rest
+    },
+    forwardedRef,
+  ) => {
+    const [checked, setChecked] = useState(false)
+    const bounceValue = useRef(new Animated.Value(1)).current
+
+    useEffect(() => {
+      setChecked(isChecked ?? false)
+    }, [isChecked])
+
+    const bounceInEffect = () => {
+      Animated.spring(bounceValue, {
+        toValue: bounceEffectIn,
+        velocity: bounceVelocityIn,
+        bounciness: bouncinessIn,
+        useNativeDriver,
+      }).start()
+    }
+
+    const bounceOutEffect = () => {
+      Animated.spring(bounceValue, {
+        toValue: bounceEffectOut,
+        velocity: bounceVelocityOut,
+        bounciness: bouncinessOut,
+        useNativeDriver,
+      }).start()
+    }
+
+    const syntheticBounceEffect = () => {
+      Animated.sequence([
+        Animated.timing(bounceValue, {
+          toValue: bounceEffectIn,
+          duration: 50,
+          useNativeDriver,
+        }),
+        Animated.spring(bounceValue, {
+          toValue: bounceEffectOut,
+          velocity: bounceVelocityOut,
+          bounciness: bouncinessOut,
+          useNativeDriver,
+        }),
+      ]).start()
+    }
+
+    const renderCheckIcon = () => {
+      const checkStatus = disableBuiltInState ? isChecked! : checked
+
+      return (
+        <Animated.View
+          testID={'icon-container'}
+          style={StyleSheet.flatten([
+            styles.iconContainer,
+            {transform: [{scale: bounceValue}]},
+            {
+              width: size,
+              height: size,
+              borderRadius: size / 4,
+              backgroundColor: checked ? fillColor : unfillColor,
+            },
+            iconStyle,
+            {opacity: disable ? disableOpacity : 1},
+          ])}>
+          <View
+            style={StyleSheet.flatten([
+              styles.innerIconContainer,
+              {width: size, height: size, borderRadius: size / 4, borderColor: fillColor},
+              innerIconStyle,
+            ])}>
+            {iconComponent ||
+              (checkStatus && (
+                <Image
+                  source={checkIconImageSource}
+                  style={StyleSheet.flatten([styles.iconImageStyle, iconImageStyle])}
+                />
+              ))}
+          </View>
+        </Animated.View>
+      )
+    }
+
+    const renderCheckboxText = () => {
+      const checkDisableTextType = typeof disableText === 'undefined'
+      return (
+        (!disableText || checkDisableTextType) &&
+        (textComponent || (
+          <View
+            style={StyleSheet.flatten([
+              styles.textContainer,
+              textContainerStyle,
+              {opacity: disable ? disableOpacity : 1},
+            ])}>
+            <Text testID="text" style={textStyle}>
+              {text}
+            </Text>
+          </View>
+        ))
+      )
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why you need to ignore this eslint rule?

Reply:
This branch don't have proper eslint config before, i will merge and remove it